src/modules/SaveSelector.ts

```diff
@@ -87,6 +87,19 @@ export default class SaveSelector {
         }
     }
 
+    static btoa(saveString: string):string {
+        return btoa(saveString.replace(/[^\u0000-\u00FF]+|%/g, (m) => encodeURI(m)));
+    }
+
+    static atob(encodeString: string):string {
+        const decodeString = atob(encodeString);
+        try {
+            return decodeURI(decodeString);
+        } catch {
+            return decodeString;
+        }
+    }
```

Comment:
I haven't had a chance to test it, but I think there is an issue with this. If we have an old save with a % in a catch filter and also a pokemon nickname, the nickname one will have been uri encoded but not the catch filter, so we error and drop into the catch block, and just return the decodeString. This leaves us with all of the nicknames etc still uri encoded.

I think we will need to always return decodeString here based on the save version in decodeString, and make an update function to handle decoding everything that was being handled individually before this version.

Reply:
This is indeed a flaw, because it is difficult to distinguish whether % is encoded or not.
So I chose a compromise solution, returning an undecoded save string to ensure that the save file can be imported successfully. This does have an effect, but I think it's very slight, only in 0.10.11 → 0.10.12, and only affects those who enter % in filters and elsewhere.

Reply:
If you really want a update function, I added it in the latest commit.
It's just a way to handle it and I'm not sure if the nesting `try...catch` block does anything to future version.
Feel free to modify my code.

If you need a save file to test, you can try the attachment in live version, prev `try...catch` version and current nesting `try...catch` version.
[[v0.10.11] PokeClicker 2023-05-22 11_39_08 .txt](https://github.com/pokeclicker/pokeclicker/files/11583075/v0.10.11.PokeClicker.2023-05-22.11_39_08.txt)